rawrequest/client_test.go

```diff
@@ -0,0 +1,130 @@
+package rawrequest
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+	stripe "github.com/stripe/stripe-go/v79"
+	_ "github.com/stripe/stripe-go/v79/testing"
+)
+
+func stubAPIBackend(testServer *httptest.Server) {
+	backend := stripe.GetBackendWithConfig(
+		stripe.APIBackend,
+		&stripe.BackendConfig{
+			LeveledLogger: &stripe.LeveledLogger{
+				Level: stripe.LevelDebug,
+			},
+			MaxNetworkRetries: stripe.Int64(0),
+			URL:               stripe.String(testServer.URL),
+		},
+	).(*stripe.BackendImplementation)
+
+	stripe.SetBackend(stripe.APIBackend, backend)
+}
+
+func TestV2PostRequest(t *testing.T) {
+	var body string
+	var path string
+	var method string
+	var contentType string
+	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		req, _ := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		body = string(req)
+		path = r.URL.RequestURI()
+		method = r.Method
+		contentType = r.Header.Get("Content-Type")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"object": "abc", "xyz": {"def": "jih"}}`))
+	}))
+
+	stubAPIBackend(testServer)
+
+	var params *stripe.RawParams
+
+	_, err := Post("/v2/abc", `{"xyz": {"def": "jih"}}`, params)
+	assert.NoError(t, err)
+
+	assert.Nil(t, params) // original params should not be modified
+	assert.Equal(t, `{"xyz": {"def": "jih"}}`, body)
+	assert.Equal(t, `/v2/abc`, path)
+	assert.Equal(t, `POST`, method)
+	assert.Equal(t, `application/json`, contentType)
+	assert.NoError(t, err)
+	defer testServer.Close()
+}
+
+func TestRawV1PostRequest(t *testing.T) {
+	var body string
+	var path string
+	var method string
+	var contentType string
+	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		req, _ := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		body = string(req)
+		path = r.URL.RequestURI()
+		method = r.Method
+		contentType = r.Header.Get("Content-Type")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"object": "abc", "xyz": {"def": "jih"}}`))
+	}))
+
+	stubAPIBackend(testServer)
+
+	var params *stripe.RawParams
+
+	_, err := Post("/v1/abc", `{"xyz": {"def": "jih"}}`, params)
```

Comment:
this isnt correct for a v1 api, right?  i.e. should be form encoded?

Reply:
the test is technically wrong, yeah. this would be a 400 on the real API, but the tests treat `content` as opaque strings. The important part is that content-type is set correctly and the response is decoded. 

That said, I can update if you'd like!

Reply:
i think it makes sense to update just from a readability standpoint; this might confuse me if I were new or unfamiliar.